### ISSUE TITLE
Tooltips Disable Scrolling in Modals

### ISSUE THREAD

**Issue by christianklemp-imt** [twbs/bootstrap#23222]
When tooltips are in a modal (triggered manually so they are constantly displayed) and the modal requires scrolling, when the cursor is over a tooltip, scrolling is disabled.

Fiddle here: https://jsfiddle.net/christianklemp_imt/2w7v77e5/

Running in Google Chrome (v60.0.3112.90) on macOS sierra (v10.12.6)

**Comment by Johann-S** [comment]
Confirmed with our last dist files : https://codepen.io/Johann-S/pen/yoVKMp?editors=1010#0

**Comment by jipexu** [comment]
is it the same case with popper.js ? as i see the example use tether.js ?

**Comment by Johann-S** [comment]
Mine use Popper.js @jipexu 😉

**Comment by jipexu** [comment]
ok

**Comment by gijsbotje** [comment]
adding `pointer-events: none;` to the tooltip enables the scrolling
https://codepen.io/gijsbotje/pen/WERNGR

**Comment by Herst** [comment]
@gijsbotje Unsupported in IE10 though, see https://caniuse.com/#feat=pointer-events (click _"Show all"_) (maybe a reason for dropping support for it, as discussed e.g. in #21416).

**Comment by patrickhlauke** [comment]
@herst don't confuse Pointer Events (the JS API) with `pointer-events` (the CSS property)

**Comment by Herst** [comment]
@patrickhlauke I know that and I meant the CSS property, maybe I should have mentioned it.
(BTW, the link to Microsoft's JS API is https://caniuse.com/#feat=pointer and this is unlikely to be ever implemented in Webkit.)

**Comment by patrickhlauke** [comment]
as editor of the pointer events spec i'm aware of Apple's political position on the matter ;)

**Comment by lucascono** [comment]
What's the problem here?

1. The tooltip is append as a direct child of the body.
2. When the modal is activated, the `.modal-open` class is added to the body tag.
3. The `.modal-open` class has the property `overflow: hidden` (deactivated scroll).
4. At the same time, the tooltip has the property `position: absolute` calculated in relation to the body, whose scrolling we have already seen that it is deactivated.
5. In short, when scrolling over the tooltip, it does not move because it has an absolute positioning on a container without scrolling. 

Some possible solutions:

1. Have all tooltips added to their corresponding direct parents. Popper.js has this configurable option. The problem here is that using the tooltip, for example, within an svg element, would not work. I tested it on the current visual test.

2. Add instructions to modal and tooltip documentation, indicating that tooltips located within modals must be properly configured. That is, indicating that the tooltip container should be the modal and not the body.

3. Add some code to the tooltips plugin so that it recognizes if any of its ancestors has the modal class and, in that case, auto-configure the container to the modal.

As there are several possible solutions, I did not want to work on this and instead put it to consideration.

**Comment by Johann-S** [comment]
Good catch @lucascono ! :+1: 
It seems it works too on SVG elements I added one on my CodePen

using `container` option fix this issue @christianklemp-imt see : https://getbootstrap.com/docs/4.0/components/tooltips/#options

and my updated CodePen : https://codepen.io/Johann-S/pen/yoVKMp?editors=1010

Should we add a special mention in our documentation ? 

/CC @patrickhlauke , @mdo, @XhmikosR

**Comment by XhmikosR** [comment]
My main concern about SVGs is that this is non-standard. (Remember the other day on Slack I mentioned it)

**Comment by XhmikosR** [comment]
I wonder what's the reasoning for not supporting SVGs with a title attribute in the root element.

**Comment by lucascono** [comment]
I thought the same as @XhmikosR. In my opinion, option 1 seems the best. I had entered the visual test to prove just that option, but casually I found the SVG element. And then I decided to pause my work.

Putting the tooltips at the same level as your triggers would greatly facilitate the development, perhaps avoid future issues, and allow the operation with other external libraries. And implementing it is relatively simple: **`document.body`** should be changed to **`$(this.element).parent()`** in the container configuration of the plugin.

And I find it more usual to place a tooltip within a modal than within an SVG.
